### DISCUSSION TITLE
Is it okay to inject the same component context into every child component?

### DISCUSSION THREAD

**Discussion by plusmobileapps** [arkivanov/Decompose#99] [Q&A] [answered]
The samples in decompose seem to have a manual dependency injection system setup always using the `ComponentContext` provided by the router `childFactory`. I was curious if it's important to use that `ComponentContext` passed from this method, or if it would be okay to just register `ComponentContext` in a DI graph and use that when creating child components? 

As I'm digging deeper into some of the source code at [RouterEntryFactoryImpl](https://github.com/arkivanov/Decompose/blob/master/decompose/src/commonMain/kotlin/com/arkivanov/decompose/router/RouterEntryFactoryImpl.kt), it seems like it might be important to actually make sure the `ComponentContext` from the `childFactory` method is being used since it is creating a new `ComponentContext` and `MergedLifecycle`?

**Comment by arkivanov** [comment, marked as answer]
Good question! It's very important to use individual contexts for each child, as mentioned in the docs (a warning at the end) - https://arkivanov.github.io/Decompose/component/overview/#adding-a-child-component-manually

**Reply by plusmobileapps** [reply]
Ahh, thanks missed that tiny warning. Made a small improvement to make this clearer to users when looking through docs https://github.com/arkivanov/Decompose/pull/101
